### PR TITLE
[0 SIZE] Add zero-size input validation for WarpCTC kernel

### PR DESCRIPTION
**Pull request by DrRyanHuang**
### PR Category

Execute Infrastructure 

### PR Types

Improvements

### Description

针对 `ctc_loss` 的 0 Size 问题做出如下改动：

**paddle/phi/kernels/impl/warpctc_kernel_impl.h**
   - 新增对 `labels`、`logits_length`、`labels_length` 三个输入的 batch_size 校验

**test/legacy_test/test_warpctc_op.py**
   - 新增 `test_dygraph_zero_size_with_padding` 单测用例

### 是否引起精度变化

否

### CODE REVIEW

**Review by zrr1999 — DISMISSED** [review]

**Review by wanghuancoder — APPROVED** [review]
LGTM

**Comment by ShigureNyako** [comment]
✅ Cherry-pick successful! Created PR: https://github.com/PaddlePaddle/Paddle/pull/78435
